gpflow/mean_functions.py

```diff
@@ -64,7 +64,7 @@ def __init__(self, A=None, b=None):
 
     @params_as_tensors
     def __call__(self, X):
-        return tf.matmul(X, self.A) + self.b
+        return tf.tensordot(X, self.A, [[-1], [0]]) + self.b
```

Comment:
is this needed because tf.matmul() does not broadcast properly ? (i.e., once tf implements properly-broadcasting matmul, would we be able to revert this change?)

Reply:
Indeed, once tensorflow/tensorflow#216 is resolved we can use matmul here. The behaviour we want works out-of-the-box with numpy's matmul:
```
In [1]: X = np.random.randn(10, 5, 4)                                           
In [2]: A = np.random.randn(4, 3)                                               
In [3]: np.matmul(X, A).shape                                                   
Out[3]: (10, 5, 3)
```